src/app/file-manager/file-download-modal/file-download-modal.component.ts

```diff
@@ -92,4 +209,68 @@ export class FileDownloadModalComponent  {
       summary: warnMessage,
     });
   }
+
+  // This function filters repeated folders when a parent
+  // is all-selected and there are sub-folders in the list
+  filterFolderItems(sortedItems): Item[] {
+
+    const folderItems = sortedItems.filter(item => item.indeterminate === false);
+
+    const singleItems = [];
+
+    for (let i = 0; i < folderItems.length;) {
+      const parent = folderItems[i];
+
+      const isDad = this.isParent(parent, folderItems[i + 1]);
+
+      if (!singleItems.includes(parent)) {
+        singleItems.push(parent);
+      }
+
+      if (isDad) {
+        folderItems.splice(i + 1, 1);
+      } else {
+        i++;
+      }
+    }
+    return singleItems;
+  }
+
+  isParent(parent: Item, child: Item): boolean {
+
+    if (parent.type === 'File') {
+      return false;
+    }
+
+    if (child === undefined) {
+      return false;
+    }
+
+    return child.path.startsWith(parent.path);
+  }
+
+  cloneItem(item: Item): Item {
+
+    const newItem: Item = new Item(
+      item.id,
+      item.name,
+      item.created,
+      item.updated,
+      item.size,
+      item.mediaLink,
+      item.workspaceName + '/',
```

Comment:
Why add `/` here? Add a comment.

Reply:
Comment added. Due to different data models for FireCloud workspaces and Bucket files, we used this trick to homogeneize  items for downloading preflight